code/ai/aicode.cpp

```diff
@@ -881,15 +927,24 @@ void ai_init()
 {
 	if ( !ai_inited )	{
 		// Do the first time initialization stuff here		
-		try
-		{
-			parse_aitbl();
-		}
-		catch (const parse::ParseException& e)
-		{
-			mprintf(("TABLES: Unable to parse '%s'!  Error message = %s.\n", "ai.tbl", e.what()));
+		free_ai_stuff();
+
+		Num_ai_classes = 0;
+		Num_alloced_ai_classes = AI_CLASS_INCREMENT;
+		Ai_classes = (ai_class*)vm_malloc(Num_alloced_ai_classes * sizeof(ai_class));
+		Ai_class_names = (char**)vm_malloc(Num_alloced_ai_classes * sizeof(char*));
+
+		parse_aitbl("ai.tbl");
+
+		parse_modular_table("*-aic.tbm", parse_aitbl);
+
+		if (class_added_in_tbm) {
+			mprintf(("One or more Ai Classes were added in a modular table. This may affect "
+					 "ai autoscaling. If that's not desired then use a tbl instead of a tbm!\n"));
 		}
```

Comment:
Not sure about this. It seems to me as any autoscaling does not happen during parse, which means the behaviour is unaffected as far as I can tell

Reply:
My understanding is that ai autoscaling is done by the index of the ai classes in the array. When adding an ai class using a tbm, even if someone intends to use it as something between Coward and Captain, that probably won't be the case just because of where it's stored. There are already methods to turn off autoscaling, but there's not really a way to change how it works (IE: Ordering the classes according to difficulty). Hence the warning.

Reply:
Ah, I see.